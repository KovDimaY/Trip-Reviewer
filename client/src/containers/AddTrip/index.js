import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addTrip, clearNewTrip } from '../../actions';

class AddTrip extends Component {
    state = {
        formdata: {
            title: '',
            author: '',
            review: '',
            duration: '',
            rating: '1',
            price: ''
        }
    };

    componentWillReceiveProps(newProps) {
        const { newtrip } = newProps.trips;
        if (newtrip) {
            newProps.history.push(`/trips/${newtrip.tripId}`);
        }
    }

    handleInput = (event) => {
        const newFormdata = {
            ...this.state.formdata
        };
        const { value, name } = event.target;

        newFormdata[name] = value;

        this.setState({
            formdata: newFormdata
        });
    }

    submitForm = (event) => {
        event.preventDefault();
        this.props.dispatch(addTrip({
            ...this.state.formdata,
            ownerId: this.props.users.login.id
        }));
    }

    componentWillUnmount() {
        this.props.dispatch(clearNewTrip());
    }

    render() {
        const {
            title, author, review,
            duration, rating, price
        } = this.state.formdata;

        return (
            <div className="rl_container article">
                <form onSubmit={this.submitForm}>
                    <h2>Add a review</h2>

                    <div className="form_element">
                        <input
                            type="text"
                            name="title"
                            placeholder="Enter title"
                            value={title}
                            onChange={this.handleInput}
                        />
                    </div>

                    <div className="form_element">
                        <input
                            type="text"
                            name='author'
                            placeholder="Enter author"
                            value={author}
                            onChange={this.handleInput}
                        />
                    </div>

                    <textarea
                        value={review}
                        name='review'
                        onChange={this.handleInput}
                    />

                    <div className="form_element">
                        <input
                            type="number"
                            name='duration'
                            placeholder="Enter duration"
                            value={duration}
                            onChange={this.handleInput}
                        />
                    </div>

                    <div className="form_element">
                        <select
                            value={rating}
                            name='rating'
                            onChange={this.handleInput}
                        >
                            <option val="1">1</option>
                            <option val="2">2</option>
                            <option val="3">3</option>
                            <option val="4">4</option>
                            <option val="5">5</option>
                        </select>
                    </div>

                    <div className="form_element">
                        <input
                            type="number"
                            name='price'
                            placeholder="Enter Price"
                            value={price}
                            onChange={this.handleInput}
                        />
                    </div>

                    <button type="submit">Add review</button>
                </form>
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        trips: state.trips
    }
}

export default connect(mapStateToProps)(AddTrip);