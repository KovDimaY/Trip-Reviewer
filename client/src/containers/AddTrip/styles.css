.add-review-container {
    text-align: center;
}
.add-review-container .form_element {
    margin-top: 20px;
}
.add-review-container h2 {
    font-size: 33px;
    font-weight: 300;
    color: #02bcd4;
}
.add-review-container input {
    font-size: 20px;
    border-radius: 3px;
    border: 1px solid #ebebeb;
    box-sizing: border-box;
    padding: 11px 31px;
    background: #f7f7f7;
    font-weight: 300;
    width: 80%;
}
.add-review-container .label {
    display: block;
    text-align: left;
    margin-left: 10%;
    color: #02bcd4;
    font-size: 21px;
    margin-bottom: 3px;
}
.add-review-container textarea {
    background: #f7f7f7;
    border: 1px solid #ebebeb;
    border-radius: 3px;
    width: 80%;
    padding: 20px;
    box-sizing: border-box;
    font-size: 14px;
}
.add-review-container select {
    width: 80%;
    background: #f7f7f7;
    font-size: 20px;
    border: 1px solid #e1e1e1;
}
.add-review-container button {
    font-size: 20px;
    padding: 10px 52px;
    background: #02bcd4;
    color: #fff;
    font-weight: 300;
    border-radius: 3px;
    border: none;
    margin-top: 50px;
    margin-bottom: 50px;
    width: 80%;
}
.add-review-container .conf_link {
    color: #8BC34A;
    font-weight: 600;
    margin-top: 20px;
}
.add-review-container .error {
    color: #FF5722;
    margin-top: 10px;
    font-weight: 600;
}
.add-review-container .delete_post {
    text-align: center;
    margin-top: 20px;
}