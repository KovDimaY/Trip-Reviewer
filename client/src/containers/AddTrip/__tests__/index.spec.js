import React from 'react';
import { create } from 'react-test-renderer';
 
import AddTrip from './../../AddTrip';
import Root from './../../../hoc/root';

const mockComponent = props => {
    return (
        <Root>
            <AddTrip {...props} />
        </Root>
    );
};
 
describe('<AddTrip />', () => {
    it('should render component', () => {
        const tree = create(mockComponent()).toJSON();
        
        expect(tree).toMatchSnapshot();
    });
});