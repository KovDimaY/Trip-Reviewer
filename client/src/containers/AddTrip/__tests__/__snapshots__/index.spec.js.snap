// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTrip /> should render component 1`] = `
<div
  className="rl_container article"
>
  <form
    onSubmit={[Function]}
  >
    <h2>
      Add a review
    </h2>
    <div
      className="form_element"
    >
      <input
        onChange={[Function]}
        placeholder="Enter title"
        type="text"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <input
        onChange={[Function]}
        placeholder="Enter author"
        type="text"
        value=""
      />
    </div>
    <textarea
      onChange={[Function]}
      value=""
    />
    <div
      className="form_element"
    >
      <input
        onChange={[Function]}
        placeholder="Enter duration"
        type="number"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <select
        onChange={[Function]}
        value=""
      >
        <option
          val="1"
        >
          1
        </option>
        <option
          val="2"
        >
          2
        </option>
        <option
          val="3"
        >
          3
        </option>
        <option
          val="4"
        >
          4
        </option>
        <option
          val="5"
        >
          5
        </option>
      </select>
    </div>
    <div
      className="form_element"
    >
      <input
        onChange={[Function]}
        placeholder="Enter Price"
        type="number"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Add review
    </button>
  </form>
</div>
`;
